internal/resolution/client/maven_registry_client.go

```diff
@@ -151,6 +151,16 @@ func (c *MavenRegistryClient) MatchingVersions(ctx context.Context, vk resolve.V
 	return resolve.MatchRequirement(vk, versions), nil
 }
 
+func (c *MavenRegistryClient) UpdateRegistries(registries []Registry) error {
+	for _, reg := range registries {
+		if err := c.api.AddRegistry(reg.URL); err != nil {
```

Comment:
do we need to clear any existing registries? `Update` implies we will overwrite all existing ones?

Reply:
We don't want to clear the existing registries - maybe this should be renamed to `AddRegistries` and have another `SetRegistries` to clear and set the registries (though not sure if this has any use case)